src/components/Register/index.jsx

```diff
@@ -0,0 +1,154 @@
+    setErrorMessage(null);
+    // e.preventDefault();
+    firebase
+      .auth()
+      .createUserWithEmailAndPassword(email, password)
+      .then((res) => {
+        setRegisterInfo(true);
+      })
+      .catch((err) => {
+        setRegisterInfo(false);
+        setErrorMessage(err.message);
+        console.error(err);
+      });
+  };
+
+  let history = useHistory();
+
+  const routeChange = () => {
+    let path = "/register";
+    history.push(path);
+  };
+
+  const addDetails = (e) => {
+    addOrganization(e);
+    handleClick();
+    routeChange();
+  };
+
+  return (
+    <div className="formInputSection" {...layout} {...tailLayout}>
+      <Form>
+        {registerinfo ? (
+          <Alert
+            message="Thank you for registration"
+            description="Success"
+            type="success"
+            className="inputSec"
+          />
+        ) : errormessage ? (
+          <p>{errormessage}</p>
+        ) : null}
+        <Form.Item>
+          <Input
+            onChange={(e) => setOrganizationName(e.target.value)}
+            value={name}
+            placeholder={t("register.registerTitle.registerTitle4")}
+            action="click"
+            type="text"
+            className="ant-input"
+          />
+        </Form.Item>
+        <Form.Item
+          value={email}
+          onChange={(e) => setEmail(e.target.value)}
+          rules={[{ required: true, message: "Please input your email!" }]}
+        >
+          <Input
+            prefix={<MailOutlined />}
+            placeholder={t("register.registerTitle.registerTitle1")}
+            action="click"
+            type="email"
+            className="ant-input"
+          />
+        </Form.Item>
+        <Form.Item
+          value={password}
+          onChange={(e) => setPassword(e.target.value)}
+          rules={[{ required: true, message: "Please input your password!" }]}
+        >
+          <Input
+            placeholder={t("register.registerTitle.registerTitle2")}
+            action="click"
+            type="password"
+            className="ant-input"
+          />
+        </Form.Item>
+        <Form.Item
+          value={password}
+          onChange={(e) => setPassword(e.target.value)}
+          rules={[{ required: true, message: "Please input your password!" }]}
+        >
+          <Input
+            placeholder={t("register.registerTitle.registerTitle2")}
+            action="click"
+            type="password"
+            className="ant-input"
+          />
+        </Form.Item>
+        <Form.Item valuePropName="checked">
+          <Checkbox type="checkbox" required>
+            {t("register.registerTitle.registerTitle3")}
+          </Checkbox>
+        </Form.Item>
+        <Form.Item>
+          {isTesting ? (
+            <button>Donate</button>
+          ) : (
+            <Button
+              style={{ background: "#2F80ED" }}
+              type="submit"
+              onClick={addDetails}
```

Comment:
lets remove the onClick function and add on Submit for the whole form in that case u will need to use e.preventDefault()

Reply:
I tried it but it's not working, i put this function on form as well.
![image](https://user-images.githubusercontent.com/61707093/99811070-b3947e80-2b55-11eb-8fff-61b127afbb8e.png)